### PR TITLE
Update flutter_tools to look for new VM service message

### PR DESCRIPTION
**Pull request by bkonyi**
The Dart SDK will soon move away from the current Observatory message:

"Observatory listening on ..."

To a new message that no longer references Observatory:

"Dart VM Service listening on ..."

This change updates all tests with mocks to check for the new message and also adds support for the new message in `ProtocolDiscovery`.

See https://github.com/dart-lang/sdk/issues/46756

### CODE REVIEW

File: packages/flutter_tools/lib/src/test/coverage_collector.dart [hunk 1]
```diff
@@ -33,9 +33,9 @@ class CoverageCollector extends TestWatcher {
   }
 
   void _logMessage(String line, { bool error = false }) {
-    if (!verbose) {
+/*    if (!verbose) {
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/test/coverage_collector.dart:36]: remove comment?

File: packages/flutter_tools/lib/src/protocol_discovery.dart [hunk 2]
```diff
@@ -39,9 +40,11 @@ class ProtocolDiscovery {
     required Logger logger,
   }) {
     const String kObservatoryService = 'Observatory';
+    const String kServicePrefixRegExp = '(?:Observatory|The Dart VM Service is)';
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/protocol_discovery.dart:43]: can this be a static or global and shared between other diffs in this PR?

Comment (bkonyi) [reply packages/flutter_tools/lib/src/protocol_discovery.dart:43]: Is there a constants file that this could live in?

Comment (bkonyi) [reply packages/flutter_tools/lib/src/protocol_discovery.dart]: Done.

File: dev/devicelab/lib/framework/runner.dart [hunk 3]
```diff
@@ -190,7 +190,7 @@ Future<TaskResult> runTask(
       .transform<String>(const LineSplitter())
       .listen((String line) {
     if (!uri.isCompleted) {
-      final Uri? serviceUri = parseServiceUri(line, prefix: 'Observatory listening on ');
+      final Uri? serviceUri = parseServiceUri(line, prefix: RegExp('(Observatory|Dart VM Service) listening on '));
```
Comment (christopherfujino) [inline dev/devicelab/lib/framework/runner.dart:193]: nit: can this use `kServicePrefixRegExp`?

**Review by christopherfujino — APPROVED** [review]
one nit, but LGTM

File: packages/flutter_tools/lib/src/protocol_discovery.dart [hunk 4]
```diff
@@ -8,7 +8,7 @@ import 'base/io.dart';
 import 'base/logger.dart';
 import 'device.dart';
 import 'device_port_forwarder.dart';
-import 'globals.dart';
+import 'globals.dart' as global;
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/protocol_discovery.dart:11]: nit:
```suggestion
import 'globals.dart' as globals;
```

**Comment by Hixie** [comment]
FWIW, I just discovered this introduced a regression in the devicelab tests: https://github.com/flutter/flutter/issues/145964

**Pull request merged**